contrib/auth/acl/controller.go

```diff
@@ -0,0 +1,673 @@
+		Prefix: swag.StringValue((*string)(params.Prefix)),
+		Amount: paginationAmount(params.Amount),
+	})
+	if c.handleAPIError(w, err) {
+		return
+	}
+
+	response := apigen.GroupList{
+		Results: make([]apigen.Group, 0, len(groups)),
+		Pagination: apigen.Pagination{
+			HasMore:    paginator.NextPageToken != "",
+			NextOffset: paginator.NextPageToken,
+			Results:    paginator.Amount,
+		},
+	}
+	for _, g := range groups {
+		response.Results = append(response.Results, apigen.Group{
+			Name:         g.DisplayName,
+			CreationDate: g.CreatedAt.Unix(),
+		})
+	}
+
+	writeResponse(w, http.StatusOK, response)
+}
+
+func (c *Controller) UpdatePassword(w http.ResponseWriter, _ *http.Request, _ apigen.UpdatePasswordJSONRequestBody, _ string) {
+	writeError(w, http.StatusNotImplemented, "Not implemented")
+}
+
+func (c *Controller) ListUserPolicies(w http.ResponseWriter, r *http.Request, userID string, params apigen.ListUserPoliciesParams) {
+	ctx := r.Context()
+	var listPolicies func(ctx context.Context, username string, params *model.PaginationParams) ([]*model.Policy, *model.Paginator, error)
+	if swag.BoolValue(params.Effective) {
+		listPolicies = c.Auth.ListEffectivePolicies
+	} else {
+		listPolicies = c.Auth.ListUserPolicies
+	}
+	policies, paginator, err := listPolicies(ctx, userID, &model.PaginationParams{
+		After:  swag.StringValue((*string)(params.After)),
+		Prefix: swag.StringValue((*string)(params.Prefix)),
+		Amount: paginationAmount(params.Amount),
+	})
+	if c.handleAPIError(w, err) {
+		return
+	}
+
+	response := apigen.PolicyList{
+		Pagination: apigen.Pagination{
+			HasMore:    paginator.NextPageToken != "",
+			NextOffset: paginator.NextPageToken,
+			Results:    paginator.Amount,
+		},
+		Results: make([]apigen.Policy, 0, len(policies)),
+	}
+	for _, p := range policies {
+		response.Results = append(response.Results, serializePolicy(p))
+	}
+	writeResponse(w, http.StatusOK, response)
+}
+
+func serializePolicy(p *model.Policy) apigen.Policy {
+	stmts := make([]apigen.Statement, 0, len(p.Statement))
+	for _, s := range p.Statement {
+		stmts = append(stmts, apigen.Statement{
+			Action:   s.Action,
+			Effect:   s.Effect,
+			Resource: s.Resource,
+		})
+	}
+	createdAt := p.CreatedAt.Unix()
+	var acl *string
+	if p.ACL.Permission != "" {
+		acl = (*string)(&p.ACL.Permission)
+	}
+	return apigen.Policy{
+		Name:         p.DisplayName,
+		CreationDate: &createdAt,
+		Statement:    stmts,
+		Acl:          acl,
+	}
+}
+
+func (c *Controller) DetachPolicyFromUser(w http.ResponseWriter, _ *http.Request, _, _ string) {
+	writeError(w, http.StatusNotImplemented, "Not implemented")
+}
+
+func (c *Controller) AttachPolicyToUser(w http.ResponseWriter, _ *http.Request, _, _ string) {
+	writeError(w, http.StatusNotImplemented, "Not implemented")
+}
+
+func (c *Controller) GetVersion(w http.ResponseWriter, _ *http.Request) {
+	writeResponse(w, http.StatusOK, apigen.VersionConfig{
+		Version: "dev",
```

Comment:
?

Reply:
It's hardcoded version of the server - we don't really need to maintain versioning in this example package